server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -868,25 +868,29 @@ private Map<String, ClusterMetadataManifest> getLatestManifestForAllClusterUUIDs
      * @return List of cluster UUIDs. The first element is the most recent cluster UUID in the chain
      */
     private List<String> createClusterChain(final Map<String, ClusterMetadataManifest> manifestsByClusterUUID, final String clusterName) {
-        final Map<String, String> clusterUUIDGraph = manifestsByClusterUUID.values()
+        final List<ClusterMetadataManifest> validClusterManifests = manifestsByClusterUUID.values()
             .stream()
+            .filter(this::isValidClusterUUID)
+            .collect(Collectors.toList());
+        final Map<String, String> clusterUUIDGraph = validClusterManifests.stream()
             .collect(Collectors.toMap(ClusterMetadataManifest::getClusterUUID, ClusterMetadataManifest::getPreviousClusterUUID));
-        final List<String> validClusterUUIDs = manifestsByClusterUUID.values()
-            .stream()
-            .filter(m -> !isInvalidClusterUUID(m) && !clusterUUIDGraph.containsValue(m.getClusterUUID()))
+        final List<String> topLevelClusterUUIDs = validClusterManifests.stream()
             .map(ClusterMetadataManifest::getClusterUUID)
+            .filter(clusterUUID -> !clusterUUIDGraph.containsValue(clusterUUID))
             .collect(Collectors.toList());
-        if (validClusterUUIDs.isEmpty()) {
+
+        if (topLevelClusterUUIDs.isEmpty()) {
```

Comment:
if validClusterManifests is non zero, then this cannot be zero right? should we add an assertion for that as well?